lib/usd/translators/shading/mtlxSymmetricShaderReader.cpp

```diff
@@ -0,0 +1,233 @@
+        },
+        fromPython);
+}
+
+TF_REGISTRY_FUNCTION(UsdMayaShaderReaderRegistry)
+{
+    // These will have to be moved to a MaterialX aware version of the symmetric shader reader.
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_lambert_surfaceshader, TrMayaTokens->lambert);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_phong_surfaceshader, TrMayaTokens->phong);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_blinn_surfaceshader, TrMayaTokens->blinn);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_place2dTexture_vector2, TrMayaTokens->place2dTexture);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->LdkND_FloatCorrect_float, TrMayaTokens->floatCorrect);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->LdkND_ColorCorrect_color4, TrMayaTokens->colorCorrect);
+    MtlxUsd_SymmetricShaderReader::RegisterReader(
+        TrMtlxTokens->MayaND_clamp_vector3, TrMayaTokens->clamp);
+};
+
+/* static */
+UsdMayaShaderReader::ContextSupport
+MtlxUsd_SymmetricShaderReader::CanImport(const UsdMayaJobImportArgs& importArgs)
+{
+    if (importArgs.GetMaterialConversion() == TrMtlxTokens->conversionName) {
+        return ContextSupport::Supported;
+    }
+
+    return ContextSupport::Unsupported;
+}
+
+MtlxUsd_SymmetricShaderReader::MtlxUsd_SymmetricShaderReader(
+    const UsdMayaPrimReaderArgs& readerArgs,
+    const TfToken&               mayaNodeTypeName)
+    : MtlxUsd_BaseReader(readerArgs)
+    , _mayaNodeTypeName(mayaNodeTypeName)
+    , _mayaShadingNodeType(
+          UsdMayaTranslatorUtil::ComputeShadingNodeTypeForMayaTypeName(mayaNodeTypeName))
+{
+}
+
+/* override */
+bool MtlxUsd_SymmetricShaderReader::Read(UsdMayaPrimReaderContext& context)
+{
+    const UsdPrim&       prim = _GetArgs().GetUsdPrim();
+    const UsdShadeShader shaderSchema = UsdShadeShader(prim);
+    if (!shaderSchema) {
+        return false;
+    }
+
+    TfToken shaderId;
+    if (!shaderSchema.GetShaderId(&shaderId)) {
+        return false;
+    }
+
+    MStatus           status;
+    MObject           mayaObject;
+    MFnDependencyNode depNodeFn;
+    if (!(UsdMayaTranslatorUtil::CreateShaderNode(
+              MString(prim.GetName().GetText()),
+              _mayaNodeTypeName.GetText(),
+              _mayaShadingNodeType,
+              &status,
+              &mayaObject)
+          && depNodeFn.setObject(mayaObject))) {
+        TF_RUNTIME_ERROR(
+            "Could not create node of type %s for shader '%s'. "
+            "Probably missing a loadPlugin.\n",
+            _mayaNodeTypeName.GetText(),
+            prim.GetPath().GetText());
+        return false;
+    }
+
+    context.RegisterNewMayaNode(prim.GetPath().GetString(), mayaObject);
+    RegisterConstructorNodes(context, mayaObject);
```

Comment:
This call is currently the only change compared to symmetricShaderReader.cpp, but I suspect handling multi-input nodes like `ramp`  will require more code to be added.